packages/gedit.rb

```diff
@@ -1,64 +1,47 @@
-require 'package'
+require 'buildsystems/meson'
 
-class Gedit < Package
+class Gedit < Meson
   description 'GNOME Text Editor'
   homepage 'https://wiki.gnome.org/Apps/Gedit'
-  version '40.1'
+  version '46.2'
   license 'GPL-2+ CC-BY-SA-3.0'
   compatibility 'x86_64 aarch64 armv7l'
   source_url 'https://gitlab.gnome.org/GNOME/gedit.git'
   git_hashtag version
-  binary_compression 'tpxz'
+  binary_compression 'tar.zst'
 
   binary_sha256({
-    aarch64: 'd00004e8fe0b7f2f9c1884f4f7657b208ac504cf9b19ef033f58559b4683588f',
-     armv7l: 'd00004e8fe0b7f2f9c1884f4f7657b208ac504cf9b19ef033f58559b4683588f',
-     x86_64: 'dc1aee9657dfa305d79fd9f4c582515886aeb6254d06794271997385d5f3060d'
+    aarch64: 'a1f180e43690a54b65de8606bf4098c190f8e51a0ffccd357af25308adf4173c',
+     armv7l: 'a1f180e43690a54b65de8606bf4098c190f8e51a0ffccd357af25308adf4173c',
+     x86_64: '8258cfcc36d014370730b4c5c9285ddbbfbe276dc6b8d7e958ee001dfd4519b7'
   })
 
-  depends_on 'amtk' # R
-  depends_on 'at_spi2_core' # R
   depends_on 'cairo' # R
+  depends_on 'desktop_file_utils' => :build
+  depends_on 'gcc_lib' # R
   depends_on 'gdk_pixbuf' # R
+  depends_on 'glibc' # R
   depends_on 'glib' # R
   depends_on 'gobject_introspection' # R
   depends_on 'gsettings_desktop_schemas' # L
   depends_on 'gspell' # R
   depends_on 'gtk3' # R
-  depends_on 'gtk_doc' => :build
-  depends_on 'gtksourceview_4' # R
+  depends_on 'harfbuzz' # R
+  depends_on 'libgedit_amtk' # R
+  depends_on 'libgedit_gtksourceview' # R
   depends_on 'libpeas' # R
   depends_on 'pango' # R
-  depends_on 'pygobject'
+  depends_on 'py3_lxml' => :build
+  depends_on 'pygobject' => :build
   depends_on 'tepl_6' # R
   depends_on 'vala' => :build
   depends_on 'yelp_tools' => :build
 
-  def self.patch
-    # Source has libgd repo as submodule
-    @git_dir = 'subprojects/libgd'
-    @git_hash = 'c7c7ff4e05d3fe82854219091cf116cce6b19de0'
-    @git_url = 'https://gitlab.gnome.org/GNOME/libgd.git'
-    FileUtils.rm_rf(@git_dir)
-    FileUtils.mkdir_p(@git_dir)
-    Dir.chdir @git_dir do
-      system 'git init'
-      system "git remote add origin #{@git_url}"
-      system "git fetch --depth 1 origin #{@git_hash}"
-      system 'git checkout FETCH_HEAD'
-    end
-  end
-
-  def self.build
-    system "meson setup #{CREW_MESON_OPTIONS} \
-    -Drequire_all_tests=false \
-    -Duser_documentation=false \
-    builddir"
-    system 'meson configure --no-pager builddir'
-    system 'ninja -C builddir'
-  end
+  meson_options '-Drequire_all_tests=false \
+    -Dgtk_doc=false \
+    -Duser_documentation=true'
 
-  def self.install
-    system "DESTDIR=#{CREW_DEST_DIR} ninja -C builddir install"
+  def self.postinstall
```

Comment:
The package [already does this](https://gitlab.gnome.org/GNOME/gedit/-/blob/da63f4c642c8ffabb0773f1ef2c2a0edadfdc3e9/meson.build#L165)

Reply:
In theory. It didn't work in practice until I ran that. Running it again to make sure that happens won't hurt anything.

Reply:
Ah this is why: The post_install scripts are skipped if a `DESTDIR` is specified in the install step. See https://github.com/mesonbuild/meson/issues/8268#issuecomment-770017342